include/experimental/__p0009_bits/extents.hpp

```diff
@@ -13,505 +13,602 @@
 
-  using rank_type = size_t;
-  using index_type = ThisIndexType;
-  static_assert(std::is_integral<index_type>::value && !std::is_same<index_type, bool>::value,
-                "std::extents requires a signed or unsigned integer as index_type parameter");
-  using size_type = make_unsigned_t<index_type>;
+  MDSPAN_INLINE_FUNCTION
+  constexpr T operator[](int r) const { return impl_t::get(r); }
+  MDSPAN_INLINE_FUNCTION
+  constexpr static T get(int r) { return impl_t::get(r); }
+  template <size_t r> MDSPAN_INLINE_FUNCTION constexpr static T get() {
+    return impl_t::template get<r>();
+  }
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t size() { return sizeof...(Values); }
+};
 
-// internal typedefs which for technical reasons are public
-  using __storage_t = detail::__partially_static_sizes_tagged<detail::__extents_tag, index_type, size_t, Extents...>;
 
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-  _MDSPAN_NO_UNIQUE_ADDRESS __storage_t __storage_;
-#else
-  using __base_t = detail::__no_unique_address_emulation<__storage_t>;
-#endif
+// ------------------------------------------------------------------
+// ------------ index_sequence_scan ---------------------------------
+// ------------------------------------------------------------------
 
-// private members dealing with the way we internally store dynamic extents
- private:
+// index_sequence_scan takes compile time values and provides get(r)
+// and get<r>() which return the sum of the first r-1 values.
 
-  MDSPAN_FORCE_INLINE_FUNCTION _MDSPAN_CONSTEXPR_14
-  __storage_t& __storage() noexcept {
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-    return __storage_;
+// Recursive implementation for get
+template <size_t R, size_t... Values> struct index_sequence_scan_impl;
+
+template <size_t R, size_t FirstVal, size_t... Values>
+struct index_sequence_scan_impl<R, FirstVal, Values...> {
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t get(size_t r) {
+    if (r > R)
+      return FirstVal + index_sequence_scan_impl<R + 1, Values...>::get(r);
+    else
+      return 0;
+  }
+  template <size_t r> MDSPAN_INLINE_FUNCTION constexpr static size_t get() {
+#if MDSPAN_HAS_CXX_17
+    if constexpr (r > R) {
+      return FirstVal + index_sequence_scan_impl<R + 1, Values...>::get(r);
+    } else {
+      return 0;
+    }
 #else
-    return this->__base_t::__ref();
+    return get(r);
 #endif
   }
-  MDSPAN_FORCE_INLINE_FUNCTION
-  constexpr __storage_t const& __storage() const noexcept {
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-    return __storage_;
+};
+
+template <size_t R, size_t FirstVal>
+struct index_sequence_scan_impl<R, FirstVal> {
+#if defined(__NVCC__) || defined(__NVCOMPILER)
+  // NVCC warns about pointless comparison with 0 for R==0 and r being const
+  // evaluatable and also 0.
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t get(size_t r) {
+    return static_cast<int64_t>(R) > static_cast<int64_t>(r) ? FirstVal : 0;
```

Comment:
Would you consider rephrasing the comparison to avoid the warning?  It's something that could come up with other compilers, given that `R` is unsigned and definitely known at compile time.
```suggestion
    return R == 0 || R <= r ? size_t(0) : FirstVal;
```